File: Intercambiar.java
import java.util.Scanner;
import java.io.InterruptedIOException;
import java.lang.Thread;
import java.io.IOException;
//=============================================================
public class Intercambiar{
	//---------------------------------------------------
	public static void main(String[] args){
		int arr[]= new int[]{1,10};
		Thread h1= new Thread(new Runnable(){
				public void run(){
					intercambiar(arr);
		}});
		//h1.start();
		arr[0]*=2;
		try{
			h1.join();
		}catch(InterruptedException e){
			e.printStackTrace();
		}
		arr[0]*=2;
		for(int i : arr){
			System.out.print(i+" ");
		}
		System.out.println();
	}
	//--------------------------------------------------

	public static void intercambiar(int []arr){
		int temp = arr[0];
		arr[0] = arr[1];
		arr[1] = temp;
	}
	//--------------------------------------------------
}

File: intercambiar.cpp
#include <iostream>
using namespace std;

void intercambiar(int& x, int& y);

int main(){
	int x=1,y=5;
	intercambiar(x, y);
	x= 2*x;
	cout<<"x: "<<x<<endl;
	cout<<"y: "<<y<<endl;
}

void intercambiar(int& x,int& y){
	int temp;
	temp = x;
	x= y;
	y = temp;
}
